src/cryptography/hazmat/backends/openssl/rsa.py

```diff
@@ -74,22 +87,32 @@ def _enc_dec_rsa(backend, key, data, padding):
     return _enc_dec_rsa_pkey_ctx(backend, key, data, padding_enum, padding)
 
 
-def _enc_dec_rsa_pkey_ctx(backend, key, data, padding_enum, padding):
+def _enc_dec_rsa_pkey_ctx(
+    backend,
+    key: typing.Union[RSAPrivateKey, RSAPublicKey],
```

Comment:
This should be typed with the concrete types, right?

Reply:
It should, yes.